converter.py

```diff
@@ -0,0 +1,183 @@
+from abc import ABC, abstractmethod, abstractproperty
+from functools import reduce
+from dataclasses import dataclass, field
+from os import path
+
+
+@dataclass
+class Geometry(ABC):
+    pass
+
+
+@dataclass(order=True, frozen=True)
+class Cylinder(Geometry):
+    sort_index: int = field(init=False)
+
+    id: str
+    radius: int = 10
+    height: int = 10
+
+    template: str = """Geometry Cyl
+    Name ScoringCylinder
+    R 0.0 10.0 {cyl_nr:d}
+    Z 0.0 30.0 {cyl_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(cyl_nr=self.radius, cyl_nz=self.height)
+
+
+@dataclass(order=True, frozen=True)
+class Mesh(Geometry):
+    sort_index: int = field(init=False)
+
+    id: str
+    x: int = 10
+    y: int = 10
+    z: int = 10
+
+    template: str = """Geometry Mesh
+    Name MyMesh_YZ
+    X -5.0  5.0    {mesh_nx:d}
+    Y -5.0  5.0    {mesh_ny:d}
+    Z  0.0  30.0   {mesh_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(mesh_nx=self.x, mesh_ny=self.y, mesh_nz=self.z)
+
+
+class BeamConfig:
+    """Class mapping of the beam.dat config file"""
+
+    beam_template: str = """
+RNDSEED      	89736501     ! Random seed
+JPART0       	2            ! Incident particle type
+TMAX0      	{energy:3.6f}   0.0  ! Incident energy; (MeV/nucl)
+NSTAT       {nstat:d}    -1 ! NSTAT, Step of saving
+STRAGG          2            ! Straggling: 0-Off 1-Gauss, 2-Vavilov
+MSCAT           2            ! Mult. scatt 0-Off 1-Gauss, 2-Moliere
+NUCRE           0            ! Nucl.Reac. switcher: 1-ON, 0-OFF
+"""
+
+    def __init__(self, energy: float = 250., nstat: int = 10000) -> None:
+        self.energy = energy
+        self.nstat = nstat
+
+    def __str__(self) -> str:
+        return self.beam_template.format(energy=self.energy, nstat=self.nstat)
+
+
+class MatConfig:
+    """Class mapping of the mat.dat config file"""
+
+    material_template: str = """MEDIUM {idx}
+ICRU {mat}
+END
+"""
+
+    def __init__(self, materials: list[int] = [276]) -> None:
+        self.materials: list[str] = materials
+
+    def __str__(self) -> str:
+        material_strings = [self.material_template.format(
+            idx=idx, mat=mat) for idx, mat in enumerate(self.materials)]
+        return "\n".join(material_strings)
+
+
+class DetectConfig:
+    """Class mapping of the detect.dat config file"""
+    detect_template = """Output
+    Filename mesh.bdo
+    Geo ScoringCylinder
+    Quantity Dose
+    """
+
+    def __init__(self, geometries: list[Geometry] = [Cylinder(), Mesh()]) -> None:
+        self.geometries: Geometry = geometries
+
+    def __str__(self):
+        detect_strings = [str(geo)
+                          for geo in self.geometries] + [self.detect_template]
+        return "\n".join(detect_strings)
+
+
+class GeoConfig:
```

Comment:
use dataclass as well

Reply:
Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f